### ISSUE TITLE
Ability to find all issues scheduled for a given day (programmatically) from Month and Quarter views

### ISSUE THREAD

**Issue by tomikall** [mailru/jira-plugins-mailrucal#58]
Currently there is no way to programmatically find out which issues are scheduled for a given day. We want to operate on a given day's issues over the JIRA REST interface using Javascript but currently this is unnecessarily difficult due to the calendar not including the required information in the DOM tree.

Within div.fc-content-skeleton the td elements in the tbody of the table do not contain the date as a data attribute like they do in the thead part. If they did, finding the right elements would be very easy indeed.

**Comment by tomikall** [comment]
Without having the date data attribute, my brute force method for finding all issues for a given day is to find all issues for a given calendar week (these are all within the same element) and narrow the selection down by querying the start timestamp (custom field) of the issues (over REST).

This is all too difficult and ugly a method to use long-term but it gets the job done when dealing with a limited number of issues. In fact, its so ugly that I shame to publicly admit that I had to do it like that :-) If you already have a better method in mind, please suggest it.

**Comment by andreymolchanov** [comment]
@tomikall, why do you need all this stuff?
Isn't it easier to request issues scheduled for a given day using JIRA REST API?
Could you please describe what you are trying to set up, maybe we can help? :-)

**Comment by tomikall** [comment]
:-) Sorry to keep you guys answering my questions. I like your calendar a lot (a bit of an understatement, really) and want to make sure it is not only good but can also be extended.

We are planning to log work on issues shown by a given filter. Our issues are shown on the calendar based on their start and finish custom fields. There are many scheduled issues for a given day and we want a quick way to log work for all of those issues at once (i.e. one day at a time). So if there are ten issues scheduled -> ten worklogs will be added.

The users will of course have to make adjustments to the worklogs but that they can do on the Tempo side where there is a really handy week view in which they can drag & drop and resize their own worklogs based on the actual time taken by each task.

Now, to answer your question regarding the REST API. I can surely request all issues for a given day using REST. But that does not help me much since there are the filters (on the right of the calendar) which define the subset of issues that we are dealing with.

This worklog-addition-in-bulk is just one example of a feature that can be added on top of your calendar. It works already but like I tried to explain above, it is not clean enough. Here's an example of how it looks:

![screenshot 2016-05-10 15 49 44](https://cloud.githubusercontent.com/assets/132639/15147048/1f05b18c-16c7-11e6-9c7b-f7d75d9b5386.png)

There is a small calendar icon which the user can use to log work for that particular day. And since the calendar is so integral part of their daily workflow, logging work directly from the calendar just makes sense. Later they review their worklogs for the whole week and send the whole bunch for approval.

**Comment by tomikall** [comment]
tl;dr
No, I don't think it is easier to use the REST API _only_ because of the filters.

**Comment by andreymolchanov** [comment]
Maybe good REST API of our calendar will be a solution? So you can get issues list for the calendar and for a given date?
I am pretty much sure that mixing presentation and logic layers won't lead us to anything good. :)

**Comment by mashintsev** [comment]
You check out our private REST API:
[JIRA-host]/rest/mailrucalendar/1.0/calendar/events/{calendarId}?start=2016-12-26&end=2017-02-06 - issues for calendar between dates
